packages/snaps-execution-environments/src/common/validation.ts

```diff
@@ -207,3 +207,28 @@ export type ExecuteSnap = RequestFunction<
   OkResponse
 >;
 export type SnapRpc = RequestFunction<SnapRpcRequestArguments, SnapRpcResponse>;
+
+/**
+ * Sanitize JSON structure.
+ *
+ * Note:
+ * This function will stringify and then parse the object provided
+ * to ensure that there are no getters which can have side effects
+ * that can cause security issues.
+ * For complete safety, this function should always be used with isValidJson
+ * (after or before) which is using Superstruct validation.
+ *
+ * @param value - JSON structure to be processed.
+ * @returns Sanitized JSON structure.
+ */
+export function sanitizeJsonStructure(value: unknown): unknown {
```

Comment:
Very good call on the `unknown` return! If you want you could might do all of the validation here 
```suggestion
export function sanitizeJsonStructure(value: unknown): unknown {
  try {
    assertStruct(value, JsonStruct);
    const sanitized = JSON.parse(JSON.stringify(value));
    assertStruct(sanitised, JsonStruct);
    return sanitized;
  }
}
```

Reply:
Alright, refactored it now.